Added examples for index routes

LalithaRajanala opened the pull request:
Added examples for index routes as mentioned in issue link: 
https://github.com/ember-learn/guides-source/issues/156

jenweber commented:
Hi @LalithaRajanala! Thanks for this PR! There are a few edits it needs before it can be merged:

- please wrap filenames and routes in backticks, like ``router.js`` and ``/posts``
- Guides URLs should be relative, so that if you click them, you're always on the same version. Here's an example. `https://guides.emberjs.com/release/routing/defining-your-routes/#toc_dynamic-segments` should be `#toc_dynamic-segments`, I think. You'll have to double check if that works. For future PRs, if you need to link to other pages, see the "When linking to topics from guide md files" section of [Contributing.md](https://github.com/ember-learn/guides-source/blob/master/CONTRIBUTING.md)
- Add blank lines surrounding code blocks
- Read it aloud - there are a couple typos. I can take a final pass to find them once you've made the above fixes

Let me know if you have any questions! Your work will help out new learners a bunch.

LalithaRajanala commented:
Thanks for the feedback and review @jenweber ! Appreciate it. I addressed the review comments and updated the pull request. Please take a look and let me know if you would like me to make any more changes.

jenweber commented:
I have reordered a couple things, fixed some typos, and changed the final example to be a bit clearer. @LalithaRajanala have a look and let me know what you think of the edits!

LalithaRajanala commented:
@jenweber : It looks good to me. Thank you ! :)

cah-danmonroe commented on guides/v3.5.0/routing/defining-your-routes.md:
````diff
@@ -187,14 +187,72 @@ Router.map(function() {
+When the user navigates to `/posts/`, the following markup will be seen:  
+
+```handlebars {data-filename=templates/posts/index.hbs} 
+<h1>This the posts template, containing headers to show on all child routes</h1>
````
```suggestion
<h1>This is the posts template, containing headers to show on all child routes</h1>
```

jenweber commented:
@cah-danmonroe thanks for the suggestions! I incorporated all of them myself, since v3.6 ships very soon. 

Thanks @LalithaRajanala for your great work!
